### ISSUE TITLE
Compiler warnings from osc/rdma

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#3056]
@hjelmn Getting some compiler warnings from GCC 5.2.0 in osc/rdma on master:

```
In file included from osc_rdma_frag.c:13:0:
osc_rdma.h:516:13: warning: 'ompi_osc_rdma_oor' defined but not used [-Wunused-function]
 static bool ompi_osc_rdma_oor (int rc)
             ^
osc_rdma.h:516:13: warning: always_inline function might not be inlinable [-Wattributes]
  CC       osc_rdma_request.lo
```

I can move the definition of `ompi_osc_rdma_oor` from `osc_rdma.h` to `osc_rdma_lock.h` to avoid the "defined but not used" warnings, but the "might not be inlinable" warnings remain.

Can you work around this, or remove the "always inline" attribute?

**Comment by hjelmn** [comment]
The inline keyword is missing.

**Comment by hjelmn** [comment]
will fix and update PRs

**Comment by jsquyres** [comment]
Fixed via #3057.  Thanks @hjelmn
